### PR TITLE
⚠️ clusterv1.ObjectMeta should only expose labels and annotations

### LINKED ISSUE #4358: clusterv1.ObjectMeta should only expose labels and annotations
The ObjectMeta type (see https://github.com/kubernetes-sigs/cluster-api/blob/5cd1132aa75f6358d66cf977164ace68ec87efe0/api/v1alpha4/common_types.go#L131) currently exposes a lot more fields that are actually used within our controllers.

In an effort to simplify, I'd like to prose we trim it down to the minimum necessary (labels and annotations).

/kind feature
/kind api-change
/milestone v0.4

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

Removes unused field from clusterv1.ObjectMeta copy.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4358 

/milestone 0.4
/assign @CecileRobertMichon @detiber

### CODE REVIEW

**Comment by vincepri** [comment]
Once this merges, we can tackle https://github.com/kubernetes-sigs/cluster-api/issues/4337

File: api/v1alpha3/common_types.go [hunk 1]
```diff
@@ -105,6 +105,8 @@ type ObjectMeta struct {
 	// Cannot be updated.
 	// More info: http://kubernetes.io/docs/user-guide/identifiers#names
 	// +optional
+	//
+	// Deprecated: This field has no function and is going to be removed in a next release.
```
Comment (fabriziopandini) [inline api/v1alpha3/common_types.go:109]: Question: should we backport the deprecation warning only to the release 0.3 branch?

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

File: api/v1alpha3/common_types.go (hunk 1 above)
Comment (vincepri) [reply api/v1alpha3/common_types.go:109]: We should, yeah

**Pull request merged**